Scripts/Runtime/Entities/Util/WorldUtil.cs

```diff
@@ -0,0 +1,230 @@
+                Debug.Assert(world.GetExistingSystem(systemGroupType) == null, $"System group cannot already exists in world {world.Name}. There is no way of knowing if a top level group for an individual world has been added to the player loop already.");
+                topLevelGroups[i] = (ComponentSystemGroup)world.CreateSystem(systemGroupType);
+
+                ScriptBehaviourUpdateOrder.AppendSystemToPlayerLoopList(topLevelGroups[i], ref playerLoop, playerLoopSystemType);
+            }
+
+            PlayerLoop.SetPlayerLoop(playerLoop);
+
+            return topLevelGroups;
+        }
+
+        /// <summary>
+        /// Sort all groups in a <see cref="World"/>.
+        /// </summary>
+        public static void SortAllGroupsInWorld(World world)
+        {
+            // This implementation is a bit overkill since calling ComponentSystemGroup.SortSystems() calls recursively down to subgroups
+            // but the way ScriptBehaviourUpdateOrder adds systems to the PlayerLoop prevents us from inspecting instances and any logic 
+            // to calculate which systems are top level (check if not a child of other systems) is likely more expensive than just overcalling SortSystems().
+            // SortSystems exits early of no sorting is required.
+            foreach (ComponentSystemBase system in world.Systems)
+            {
+                (system as ComponentSystemGroup)?.SortSystems();
+            }
+        }
+
+        private static readonly (Type PlayerLoopSystemType, Type SystemGroupType)[] s_MultiWorldTopLevelGroupTypes = new[]{
+            (typeof(PostInitialization), typeof(EndInitializationCommandBufferSystemGroup)),
+            (typeof(PostUpdate), typeof(EndSimulationCommandBufferSystemGroup)),
+        };
+        /// <summary>
+        /// Optimizes a <see cref="World"/>'s <see cref="ComponentSystemGroup"/>s for multi-world applications.
+        /// Groups end command buffers into their own <see cref="PlayerLoop"/> phase just after their default phase. 
+        /// (Ex: <see cref="Update"/> -> <see cref="PostUpdate"/>)
+        /// This allows end command buffers for all worlds to be evaluated after all worlds have scheduled their work for the phase.
+        /// The result is that other worlds can compute their jobified work while one world is executing its end command buffer on the main thread.
+        /// </summary>
+        /// <param name="world">The world instance to optimize.</param>
+        /// <remarks>
+        /// This method doesn't move <see cref="EndFixedStepSimulationEntityCommandBufferSystem" because it's embedded in the <see cref="SimulationSystemGroup"/>.
+        /// making it impractical to group with the other worlds. There shouldn't be any expensive work in that command buffer anyway.
+        /// </remarks>
+        public static void OptimizeWorldForMultiWorldInCurrentPlayerLoop(World world)
```

Comment:
This is a super smart way to handle this. I really like the thought in knowing we have main thread work we must do but figuring out how to arrange it so that everyone else gets a chance to load up the workers before the main thread work goes on.